pkg/syncer/cnsoperator/controller/csinodetopology/csinodetopology_controller.go

```diff
@@ -0,0 +1,336 @@
+	err = c.Watch(&source.Kind{Type: &csinodetopologyv1alpha1.CSINodeTopology{}}, &handler.EnqueueRequestForObject{}, pred)
+	if err != nil {
+		log.Errorf("Failed to watch for changes to CSINodeTopology resource with error: %+v", err)
+		return err
+	}
+	return nil
+}
+
+// blank assignment to verify that ReconcileCSINodeTopology implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileCSINodeTopology{}
+
+// ReconcileCSINodeTopology reconciles a CSINodeTopology object
+type ReconcileCSINodeTopology struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client        client.Client
+	scheme        *runtime.Scheme
+	clusterFlavor cnstypes.CnsClusterFlavor
+	configInfo    *cnsconfig.ConfigurationInfo
+	recorder      record.EventRecorder
+}
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// Note: The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileCSINodeTopology) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	log := logger.GetLogger(ctx)
+
+	// Fetch the CSINodeTopology instance
+	instance := &csinodetopologyv1alpha1.CSINodeTopology{}
+	err := r.client.Get(ctx, request.NamespacedName, instance)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Infof("CSINodeTopology resource with name %q not found. Ignoring since object must have been deleted.", request.Name)
+			return reconcile.Result{}, nil
+		}
+		log.Errorf("Failed to fetch the CSINodeTopology instance with name: %q. Error: %+v", request.Name, err)
+		// Error reading the object - return with err
+		return reconcile.Result{}, err
+	}
+
+	// Get NodeVM instance
+	nodeID := instance.Spec.NodeID
+	nodeManager := node.GetManager(ctx)
+	// NOTE: NodeID is set to NodeName for now. Will be changed to NodeUUID in future.
+	nodeVM, err := nodeManager.GetNodeByName(ctx, nodeID)
+	if err != nil {
+		// If nodeVM not found, ignore reconcile call
+		log.Warnf("Ignoring update to CSINodeTopology CR with name %s as corresponding NodeVM seems to be deleted. "+
+			"Error: %v", nodeID, err)
+		_ = updateCRStatus(ctx, r, instance, csinodetopologyv1alpha1.CSINodeTopologyError,
+			fmt.Sprintf("Failed to find NodeVM with ID: %q", nodeID))
+		return reconcile.Result{}, nil
+	}
+
+	// Initialize backOffDuration for the instance, if required.
+	backOffDurationMapMutex.Lock()
+	var timeout time.Duration
+	if _, exists := backOffDuration[instance.Name]; !exists {
+		backOffDuration[instance.Name] = time.Second
+	}
+	timeout = backOffDuration[instance.Name]
+	backOffDurationMapMutex.Unlock()
+
+	// Fetch vsphere config
+	cfg, err := common.GetConfig(ctx)
```

Comment:
The `ReconcileCSINodeTopology` struct has a `ConfigInfo` member variable. Can that be used here instead of reinitializing?

Reply:
Good point, removed the call.